apple/RNSVGUIKit.h

```diff
@@ -8,7 +11,11 @@
 #define RNSVGColor UIColor
 #define RNSVGPlatformView UIView
 #define RNSVGTextView UILabel
+#ifdef RCT_NEW_ARCH_ENABLED
+#define RNSVGView RCTViewComponentView
```

Comment:
Did you test it on new arch on `iOS`? What about e.g. such code: https://github.com/software-mansion/react-native-svg/blob/3b360a2b58a6bbc9541f81ee6c7150ca316d583f/apple/ViewManagers/RNSVGSvgViewManager.mm#L16, should we expect `RCTViewComponentView` there? Shouldn't it be rather `UIView` ? Can we narrow the class this way?

Reply:
As I understand we need _props value from RCTViewComponentView, because it does not exist in UIView, and we use that in RNSVGPattern.mm and RNSVGPath.mm.

Reply:
Why do we need `_props` in the file I linked?

Reply:
I mean there was always just `UIView` passed there: https://github.com/software-mansion/react-native-screens/blob/2cd05a8e39b81ec0458aba9e733c5a8a95a2d1d6/ios/RNSScreen.mm#L1489 and right now it will get `RCTViewComponentView` if I understand the code correctly, yeah?

Reply:
[That](https://github.com/software-mansion/react-native-svg/pull/2341/files#diff-f69c74ee636ed35a8d1db786a7f2446e211e3566a7155bbbedb3b61d1ea31864R16) changes should answer your question. Now we specify the returned type from view.
here:
```
@interface RNSVGView :
#ifdef RCT_NEW_ARCH_ENABLED
    RCTViewComponentView
#else
    RCTUIView
#endif // RCT_NEW_ARCH_ENABLED
```
- macOS:
 Fabric: `RCTViewComponentView` extends from  `RCTUIView` -> `RCTPlatformView` -> `RCTPlatformView NSView`.
Paper: `RCTUIView` extends from `RCTPlatformView` -> -> `RCTPlatformView NSView`.

- IOS:
```
#ifdef RCT_NEW_ARCH_ENABLED
#define RNSVGView RCTViewComponentView
#else
#define RNSVGView UIView
#endif // RCT_NEW_ARCH_ENABLED
```
Fabric: `RCTViewComponentView` extends from `UIView`
Paper: `UIView`

Reply:
Here is an article that shows the problem: https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science). E.g. here: https://github.com/software-mansion/react-native-svg/blob/26c4c73400d36513de61f00b7297104c6c7d960d/apple/Elements/RNSVGClipPath.mm#L63 you will now assume on new arch that `self.subviews` has elements of kind `RCTViewComponentView` which is certainly not true and it seems weird that it even compiles.